### PR TITLE
Permissions V 0.1. Still need feedback on the functionality. Create e…

### PR DESCRIPTION
**Pull request by XennMaestro**
…vent related functions in the Utility forlder. Modified Events to work with the new functionality.

### CODE REVIEW

**Comment by XennMaestro** [comment]
Not 100% sure how the Permissions work, but i tired to replicate Brandon's conference page. So it should work similarly.

File: app/Utility/PermissionNames.php [hunk 1]
```diff
@@ -23,4 +23,12 @@ public static function ConferenceRegistrationApproval($confId) {
     public static function ConferenceInfoEdit($confId) {
         return "conference-info-edit." . $confId;
     }
+
+    public static function EventCreate() {
```
Comment (brandonson) [inline app/Utility/PermissionNames.php]: This permission is redundant and not quite right - use ConferenceEventCreate, as that's per-conference.  With this, anyone who can create an event can create an event for any conference.

Then, in whatever function creates the event, do something like this:

```
if (!Entrust::can(PermissionNames::ConferenceEventCreate($conferenceId))) {
    return response("", 403);
}
```

**Comment by brandonson** [comment]
Quick rundown on permissions - You have a hierarchy where each user has (0 or more) roles, and each role has (0 or more) permissions.  (Different roles can share the same or some of the same permissions)

To grant the current user some role, use `Auth::user()->attachRole($role)` or some equivalent.  Same idea with roles and permissions: `$role->attachPermission($permission)`.

Then, to check if the current user has a certain permission, you can use `Entrust::can("permissionName")` (or, in our case, `Entrust::can(PermissionNames::SomePermissionName(<id if needed>))`)

**Comment by brandonson** [comment]
Apart from the inline comment, a couple general comments about the PR:

Check indentation.  You've got issues in a few places where you've got 3 spaces instead of 4, and some places where you need to indent another level and don't.

Other than that, I think the only thing is to base this off my permissions branch instead of adding new permissions, since they'll just conflict. (going to merge that right now though, so rebasing onto/merging in the new master should work as well).

**Pull request closed without merging**